routing/src/interfaces/interface.rs

```diff
@@ -217,25 +248,26 @@ impl Interface {
     //////////////////////////////////////////////////////////////////
     /// Add (assign) an IP address to an [`Interface`]
     //////////////////////////////////////////////////////////////////
-    pub fn add_ifaddr(&mut self, ifaddr: &IfAddress) {
-        self.addresses.insert(*ifaddr);
+    pub fn add_ifaddr(&mut self, ifaddr: IfAddr) {
+        self.addresses.insert(ifaddr);
     }
 
     //////////////////////////////////////////////////////////////////
     /// Del (unassign) an IP address from an [`Interface`]
     //////////////////////////////////////////////////////////////////
-    pub fn del_ifaddr(&mut self, ifaddr: &IfAddress) {
-        self.addresses.remove(ifaddr);
+    pub fn del_ifaddr(&mut self, ifaddr: IfAddr) {
```

Comment:
this observation is technically out of scope, but I don't see the point of hiding the return value here.  Shouldn't we be able to detect that the address was in fact removed?